train.py

```diff
@@ -160,9 +162,14 @@ def initialize_trainer(epochs, patience, model_name, l_rate, weight_decay, warmu
     parser.add_argument('--epochs', dest='epochs', type=int, default=50)
     parser.add_argument('--patience', dest='patience', type=int, default=10)
     parser.add_argument('--batch-size', dest='batch_size', type=int, default=64)
-    parser.add_argument('--lr', dest='l_rate', type=float, default=0.01)
-    parser.add_argument("--w-decay", dest='w_decay', type=float, default=2e-3,
-                        help="Weight decay for L2 regularization of optimizer AdamW")
+    parser.add_argument('--lr-enc', dest='l_rate_enc', type=float, default=0.01,
+                        help="Encoder learning rate.")
+    parser.add_argument('--lr-cl', dest='l_rate_cl', type=float, default=0.01,
```

Comment:
So this might be ugly to implement but would it be possible to have the default values as -1 and then later check that if the value is <0 then just make it equal to the encoder params? Just so if we don't want to separate it we can do it easily